coderd/aitasks.go

```diff
@@ -590,3 +597,209 @@ func (api *API) taskDelete(rw http.ResponseWriter, r *http.Request) {
 	// Delete build created successfully.
 	rw.WriteHeader(http.StatusAccepted)
 }
+
+// taskSend submits task input to the tasks sidebar app by dialing the agent
+// directly over the tailnet. We enforce ApplicationConnect RBAC on the
+// workspace and validate the sidebar app health.
+func (api *API) taskSend(rw http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	idStr := chi.URLParam(r, "id")
+	taskID, err := uuid.Parse(idStr)
+	if err != nil {
+		httpapi.Write(ctx, rw, http.StatusBadRequest, codersdk.Response{
+			Message: fmt.Sprintf("Invalid UUID %q for task ID.", idStr),
+		})
+		return
+	}
+
+	var req codersdk.TaskSendRequest
+	if !httpapi.Read(ctx, rw, r, &req) {
+		return
+	}
+	if req.Input == "" {
+		httpapi.Write(ctx, rw, http.StatusBadRequest, codersdk.Response{
+			Message: "Task input is required.",
+		})
+		return
+	}
+
+	type messagePayload struct {
+		Content string `json:"content"`
+		Type    string `json:"type"`
+	}
+	payload, err := json.Marshal(messagePayload{Content: req.Input, Type: "user"})
+	if err != nil {
+		httpapi.Write(ctx, rw, http.StatusInternalServerError, codersdk.Response{
+			Message: "Internal error encoding request payload.",
+			Detail:  err.Error(),
+		})
+		return
+	}
+
+	if err = api.authAndDoWithTaskSidebarAppClient(r, taskID, func(ctx context.Context, client *http.Client, appURL *url.URL) error {
+		// Build the request to the agent local app (we expect agentapi on the other side).
+		appReqURL := appURL
+		appReqURL.Path = path.Join(appURL.Path, "message")
+
+		newReq, err := http.NewRequestWithContext(ctx, http.MethodPost, appReqURL.String(), bytes.NewReader(payload))
```

Comment:
I think we need to also check if the agentapi instance is in a 'stable' state. IIRC it will reject messages if the thing backing it is doing stuff.

See: https://github.com/coder/coder/pull/19940/files#diff-1caf938e6648581b1f63b886c1f141d1dbc98827102088ffeb0c2f8825f2fe7cR120-R168